Assets/Scripts/Model/Weapon.cs

```diff
@@ -1,29 +1,37 @@
 ﻿using UnityEngine;
 using System.Collections;
+using System.Collections.Generic;
 
 public class Weapon : MonoBehaviour, CanUpgrade
 {
     public float baseDamage;
-    public float baseRadius;
     public float baseRange;
     public float baseCooldown;
+	public float upgradeFactor;
+	public AudioClip shootSound;
 
-    // TODO change all these to private
-    public float currentDamage; 
-    public float currentRadius;
-    public float currentRange;
-    public float currentCooldown;
-    //
-
-    public float upgradeFactor;
+    private float currentDamage;
+    private float currentRange;
+    private float currentCooldown;
+	private List<CanReceiveDamage> targets;
```

Comment:
Since it's an empty list for both ... it's ok if Weapon has it. Unit and Tower, since they are the ones that discover them, pass them to the list of targets of their weapon.